From 939a0dd6329e5f1270460c01d97ca7e453f77b4b Mon Sep 17 00:00:00 2001
From: torvic9 <torvic9@mailbox.org>
Date: Sun, 17 Jan 2021 22:16:42 +0100
Subject: [PATCH] allow cpufreq ondemand with muqss and projectc

This should enable the ondemand cpufreq governor when using MuQSS or 
ProjectC schedulers on non-intel_pstate systems.

Signed-off-by: torvic9 <torvic9@mailbox.org>
---
 drivers/cpufreq/Kconfig | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/cpufreq/Kconfig b/drivers/cpufreq/Kconfig
index 85de313..78f6eb3 100644
--- a/drivers/cpufreq/Kconfig
+++ b/drivers/cpufreq/Kconfig
@@ -39,8 +39,9 @@ choice
 	default CPU_FREQ_DEFAULT_GOV_USERSPACE if ARM_SA1100_CPUFREQ || ARM_SA1110_CPUFREQ
 	default CPU_FREQ_DEFAULT_GOV_SCHEDUTIL if ARM64 || ARM
 	default CPU_FREQ_DEFAULT_GOV_SCHEDUTIL if X86_INTEL_PSTATE && SMP
 	default CPU_FREQ_DEFAULT_GOV_PERFORMANCE
+	default CPU_FREQ_DEFAULT_GOV_ONDEMAND if (SCHED_ALT || SCHED_MUQSS) && !(X86_INTEL_PSTATE)
 	help
 	  This option sets which CPUFreq governor shall be loaded at
 	  startup. If in doubt, use the default setting.
 
@@ -70,9 +71,9 @@ config CPU_FREQ_DEFAULT_GOV_USERSPACE
 	  to enable the userspace governor manually.
 
 config CPU_FREQ_DEFAULT_GOV_ONDEMAND
 	bool "ondemand"
-	depends on !(X86_INTEL_PSTATE && SMP)
+	depends on (SCHED_ALT || SCHED_MUQSS) && !(X86_INTEL_PSTATE)
 	select CPU_FREQ_GOV_ONDEMAND
 	select CPU_FREQ_GOV_PERFORMANCE
 	help
 	  Use the CPUFreq governor 'ondemand' as default. This allows
@@ -145,8 +146,9 @@ config CPU_FREQ_GOV_USERSPACE
 	  If in doubt, say Y.
 
 config CPU_FREQ_GOV_ONDEMAND
 	tristate "'ondemand' cpufreq policy governor"
+	default y if SCHED_ALT || SCHED_MUQSS
 	select CPU_FREQ_GOV_COMMON
 	help
 	  'ondemand' - This driver adds a dynamic cpufreq policy governor.
 	  The governor does a periodic polling and 
-- 
2.30.0

